src/core/zowe/core_for_zowe_sdk/zosmf_profile2.py

```diff
@@ -0,0 +1,302 @@
+"""Zowe Python Client SDK.
+
+This program and the accompanying materials are made available under the terms of the
+Eclipse Public License v2.0 which accompanies this distribution, and is available at
+
+https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zowe Project.
+"""
+
+import base64
+import os.path
+import re
+import warnings
+from typing import Union
+
+import jsonc
+
+from .constants import constants
+from .exceptions import ProfileNotFound, SecureProfileLoadFailed
+
+HAS_KEYRING = True
+try:
+    import keyring
+except ImportError:
+    HAS_KEYRING = False
+
+
+class ProfileManager:
+    """
+    Class used to represent a Zowe z/OSMF profile.
+
+    Description
+    -----------
+    This class is only used when there is already a Zowe z/OSMF profile created
+    and the user opted to use the profile instead of passing the credentials directly
+    in the object constructor.
+
+    Attributes
+    ----------
+    profile_name: str
+        Zowe z/OSMF profile name
+    """
+
+    def __init__(self):
+        self._config_dir = None
+        self._config_filename = "zowe.config.json"
+        self._config_filepath = None
+
+    @property
+    def config_dir(self) -> Union[str, None]:
+        """Returns the folder path to where the Zowe z/OSMF Team Profile Config files are located."""
+        return self._config_dir
+
+    @config_dir.setter
+    def config_dir(self, dirname: str) -> None:
+        """Set directory/folder path to where Zowe z/OSMF Team Profile Config files are located"""
+        if os.path.isdir(dirname):
+            self._config_dir = dirname
+        else:
+            raise FileNotFoundError(f"given path {dirname} is not valid")
+
+    @property
+    def config_filename(self) -> str:
+        """Return the filename for Zowe z/OSMF Team Profile Config"""
+        return self._config_filename
+
+    @config_filename.setter
+    def config_filename(self, filename: str) -> None:
+        """Set filename for the Zowe z/OSMF Team Profile Config
+
+        If config_dir is set, we check for its existence in that directory,
+        If config_dir is not set, we rely on autodiscover_config_dir() while
+        loading profile to check the existence of the file and raise exceptions
+        """
+        if self._config_dir is not None:
+            path = os.path.join(self._config_dir, filename)
+
+            if os.path.isfile(path):
+                self._config_filename = filename
+            else:
+                raise FileNotFoundError(
+                    f"file not found on path {path} or set directory path first"
+                )
+        else:
+            self._config_filename = filename
```

Comment:
Instead of the `config_filename` getter/setter, I would suggest the following:
* The constructor can take an `appname` argument that defaults to `"zowe"` (`def __init__(self, appname="zowe"):`)
* `config_appname` can be a read-only property that returns the app name given to the constructor
* `config_filename` can be a read-only property like `config_filepath` that returns `{self.appname}.config.json`
* (in a future PR) `config_user_filename` can be a read-only property that returns `{self.appname}.config.user.json`

These behaviors match the Config class in the Node.js SDK:
https://github.com/zowe/imperative/blob/master/packages/config/src/Config.ts#L52